pipeline/00-ingest.R

```diff
@@ -296,33 +296,92 @@ training_data_clean <- training_data_w_hie %>%
     across(where(bit64::is.integer64), \(x) as.numeric(x))
   ) %>%
   # Get a count of the number of sales that have occurred in the last n years
+  # and the maximum price of the most recent valid sale in the last n years,
+  # both EXCLUDING the current sale
   left_join(
     left_join(
       training_data %>%
         select(meta_pin, meta_sale_document_num, meta_sale_date),
       training_data %>%
         filter(!sv_is_outlier) %>%
-        select(meta_pin, meta_sale_date),
+        select(meta_pin, meta_sale_date, meta_sale_price) %>%
+        mutate(meta_sale_price = as.numeric(meta_sale_price)),
       by = "meta_pin",
       relationship = "many-to-many"
     ) %>%
-      # as.duration(1) excludes the same sale from being identified as within
-      # 3 years of itself
-      mutate(within_n_years = between(
-        meta_sale_date.x - meta_sale_date.y,
-        as.duration(1),
-        as.duration(years(params$input$n_years_prior))
-      )) %>%
-      # Distinct is necessary because of multicard sales
-      distinct() %>%
-      summarise(
+      # as.duration(1) excludes the target sale from being identified as within
+      # N years of itself
+      mutate(
+        within_n_years = between(
+          meta_sale_date.x - meta_sale_date.y,
+          as.duration(1),
+          as.duration(years(params$input$n_years_prior))
+        ),
+        price_within_n_years = ifelse(within_n_years, meta_sale_price, NA_real_)
+      ) %>%
+      # Distinct is necessary because of multi-card sales
+      distinct(
+        meta_pin,
+        meta_sale_date.x,
+        meta_sale_date.y,
+        .keep_all = TRUE
+      ) %>%
+      mutate(price_within_n_years = replace_na(price_within_n_years, 0)) %>%
+      summarize(
         meta_sale_count_past_n_years = as.numeric(
           sum(within_n_years, na.rm = TRUE)
         ),
+        meta_sale_price_past_n_years = max(price_within_n_years, na.rm = TRUE),
```

Comment:
**[Question, non-blocking]** Why do we want the max price in the past N years as opposed to the latest price? Also, do we need `na.rm = TRUE` here given the `replace_na` call on line 329?

Reply:
No definitely not technically necessary